source/pcic/client.py

```diff
@@ -406,9 +411,14 @@ def request_last_image_taken_deserialized(self, image_id=1, datatype='ndarray'):
             results.setdefault(counter, []).append(header)
             # append image
             image_hex = data[header['HEADER_SIZE']:header['CHUNK_SIZE']]
-            if datatype == 'ndarray':
+            chunk_type = int(header['CHUNK_TYPE'])
+            if datatype == 'ndarray' and ChunkType.JPEG_IMAGE:
                 image = mpimg.imread(io.BytesIO(image_hex), format='jpg')
                 results[counter].append(image)
+            elif image_id == 2 and chunk_type == ChunkType.MONOCHROME_2D_8BIT:
+                image = np.frombuffer(image_hex, dtype=np.uint8)\
+                    .reshape((header["IMAGE_HEIGHT"], header["IMAGE_WIDTH"]))
+                results[counter].append(image)
```

Comment:
The case here does not distinguish what `datatype` is requested.

I would structure the whole if statement like this:
```
    if datatype == 'ndarray':
        if image_id == 1:
            # Check that we have received chunk type JPEG_IMAGE
            # Convert jpeg data to image data
        elif image_id == 2:
            # Check that we have received chunk type MONOCHROME_2D_8BIT
            # Read pixel data and reshape to width/height
        else:
            # Invalid/unsupported image id
    elif datatype == 'bytes':
        # return plain bytes  data
    else:
        # Invalid/unsupported datatype
```

Reply:
fixed in f2247834615e6fe0fa846a68dc16ffb50526cc14